File: dev/test_data/main/golang/expected/implementation_specific/input/snippets/Verification/has_balanced_brackets.go
// Check that the square brackets in `text` are balanced.
func HasBalancedBrackets(text string) bool {
	depth := 0
	for _, character := range text {
		if character == '[' {
			depth++
		} else if character == ']' {
			depth--
			if depth < 0 {
				return false
			}
		}
	}

	return depth == 0
}

File: dev/test_data/main/typescript/expected/implementation_specific/input/snippets/Verification/has_balanced_brackets.ts
/**
 * Check that the square brackets in `text` are balanced.
 *
 * @param text - to be verified
 * @returns `true` if the check passes
 */
export function hasBalancedBrackets(text: string): boolean {
  let depth = 0;
  for (const character of text) {
    if (character === "[") {
      depth++;
    } else if (character === "]") {
      depth--;
      if (depth < 0) {
        return false;
      }
    }
  }

  return depth === 0;
}
